core/src/main/java/org/apache/iceberg/SnapshotManager.java

```diff
@@ -50,25 +51,34 @@ protected ManageSnapshots self() {
 
   @Override
   protected String operation() {
-    // snapshotOperation is used by SnapshotProducer when building and writing a new snapshot for cherrypick
-    Preconditions.checkNotNull(snapshotOperation, "[BUG] Detected uninitialized operation");
+    //TODO: We are not setting this var during rollbacks. What should its value be during rollbacks?
```

Comment:
From the conversation IIUC, looks like this TODO can be removed and the `Precondition` check can be re-instated?  or is there still a concern here?